src/cloudcasting/types.py

```diff
@@ -0,0 +1,21 @@
+__all__ = (
+    "SingleArray",
+    "BatchArray",
+    "InputArray",
+    "TimeArray",
+    "ForecastArray",
+    "SingleForecastArray",
+)
+
+import numpy as np
+import numpy.typing as npt
+from jaxtyping import Float as Float32
+
+# Type aliases for clarity + reuse
+Array = npt.NDArray[np.float32]  # the type arg is ignored by jaxtyping, but is here for clarity
+SingleArray = Float32[Array, "channels time height width"]
```

Comment:
Should this have a more specific name, e.g. `SingleInputArray`? What's it an array of?